sigllm/primitives/prompting/huggingface.py

```diff
@@ -120,31 +130,63 @@ def detect(self, X, **kwargs):
         max_tokens = input_length * float(self.anomalous_percent)
         all_responses, all_generate_ids = [], []
 
+        # Prepare the one-shot example if provided
+        one_shot_message = ''
+        if normal is not None:
+            one_shot_message = PROMPTS['one_shot_prefix'] + normal + '\n\n'
+
         for text in tqdm(X):
             system_message = PROMPTS['system_message']
-            user_message = PROMPTS['user_message']
-            message = ' '.join([system_message, user_message, text, '[RESPONSE]'])
-
-            input_length = len(self.tokenizer.encode(message[0]))
+            if self.restrict_tokens:
+                user_message = PROMPTS['user_message']
+            else:
+                user_message = PROMPTS['user_message_2']
+
+            # Combine messages with one-shot example if provided
+            message = ' '.join([
+                system_message,
+                one_shot_message,
+                user_message,
+                text,
+                '[RESPONSE]',
+            ])
+
+            input_length = len(self.tokenizer.encode(message))
```

Comment:
this variable is not used since the code is changed to find output after `[RESPONSE]`. We should remove it